pkg/reconciler/revision/reconcile_resources.go

```diff
@@ -66,36 +66,6 @@ func (c *Reconciler) reconcileDeployment(ctx context.Context, rev *v1alpha1.Revi
 		}
 	}
 
-	// If a container keeps crashing (no active pods in the deployment although we want some)
-	if *deployment.Spec.Replicas > 0 && deployment.Status.AvailableReplicas == 0 {
-		pods, err := c.KubeClientSet.CoreV1().Pods(ns).List(metav1.ListOptions{LabelSelector: metav1.FormatLabelSelector(deployment.Spec.Selector)})
-		if err != nil {
-			logger.Errorf("Error getting pods: %v", err)
-		} else if len(pods.Items) > 0 {
-			// Arbitrarily grab the very first pod, as they all should be crashing
-			pod := pods.Items[0]
-
-			// Update the revision status if pod cannot be scheduled(possibly resource constraints)
-			// If pod cannot be scheduled then we expect the container status to be empty.
-			for _, cond := range pod.Status.Conditions {
-				if cond.Type == corev1.PodScheduled && cond.Status == corev1.ConditionFalse {
```

Comment:
Looks like this check got dropped in the rebase.

Reply:
Yes, this check was introduced after my prototype.  I wanted to move ContainerHealthy check from revision to PA first, then moving ResourceAvailable condition as well.  I can go ahead and move ResourceAvailable check to PA now too, if that makes sense?

Reply:
Moved this check to PodAutoscaler, with a new condition PodsHealthy.